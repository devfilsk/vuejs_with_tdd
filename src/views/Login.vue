<script>
export default {
    name: 'Login', 
    data() {
        return {
            email: 'admin@goscore.com.br',
            password: '6f2a3s2t',
        }
    },
    methods: {
        submitted() {
            
        }
    }
}
</script>

<template>
    <div class="content">
        <!-- <form @submit.prevent="$emit('submitted', { email, password })" class="login-content"> -->
        <form @submit.prevent="submitted()" class="login-content">
            <h1>Login</h1>
            <input type="text" class="email" v-model="email">
            <input type="text" class="password" v-model="password">
            <button type="submit" class="btn-login">Entrar</button>
        </form>
    </div>
</template>

<style lang="scss">
    .content{
        width: 100%;
        height: 100vh;
        display: flex;
        flex-direction: column;
        align-content: center;
        align-items: center;
        justify-content: center;
    }
    .login-content{
        width: 350px;
        height: 250px;
        display: flex;
        color: #FFF;
        flex-direction: column;
        padding: 20px;
        border-radius: 5px;
        background-color: rgba(255, 255, 255, .1);
        input {
            border: none;
            border-radius: 4px;
            padding: 10px 10px;
            margin-bottom: 15px;
        }
        button {
            border: none;
            padding: 10px;
            border-radius: 4px;
            cursor: pointer;
        }
    }

</style>