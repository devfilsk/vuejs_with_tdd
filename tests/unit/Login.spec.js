import Login from '@/views/Login';
import { shallowMount } from '@vue/test-utils';

describe('login', () => {
    let state;

    const build = () => {
        const wrapper = shallowMount(Login);

        return {
            wrapper,
            email: () => wrapper.find('input.email'),
            password: () => wrapper.find('input.password'),
            buttonLogar: () => wrapper.find('.btn-login')            
        }
    }

    it('should render the login component', async () => {
        const { wrapper, email, password } = build();

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render main child components', async () => {
        const { email, password, buttonLogar } = build();

        // asserts
        expect(email().exists()).toBe(true);
        expect(password().exists()).toBe(true);
        expect(buttonLogar().exists()).toBe(true);
    })

    it('should call "submited" event when submitting form',  () => {
        const { wrapper, email, password, buttonLogar } = build();

        const emailUser = 'admin@goscore.com.br';
        const passwordUser = '6f2a3s2t';

        // arrange
        email().element.value = emailUser;
        password().element.value = passwordUser;

        // act 
        email().trigger('input');
        password().trigger('input');
        buttonLogar().trigger('click');
        buttonLogar().trigger('submit');

        expect(wrapper.emitted().submitted[0]).toEqual([{ email: emailUser, password: passwordUser }])

    })
})