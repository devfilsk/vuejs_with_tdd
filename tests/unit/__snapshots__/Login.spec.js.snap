// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login should render the login component 1`] = `
<div class="content">
  <form class="login-content">
    <h1>Login</h1> <input type="text" class="email"> <input type="text" class="password"> <button class="btn-login">Entrar</button>
  </form>
</div>
`;
